### PR TITLE
Add tests for DeadlinePolicyProcessor

### PR DESCRIPTION
**Pull request by Raibo**
https://github.com/happy-travel/agent-app-project/issues/122

### CODE REVIEW

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        {
+
+            var originalDeadline = new Deadline(FromDate, Policies);
+            var checkIn = FromDate;
+            var shiftSpan = TimeSpan.FromDays(10);
+            var settings = new CancellationPolicyProcessSettings { PolicyStartDateShift = shiftSpan };
+
+            var shiftedDeadline = DeadlinePolicyProcessor.Process(originalDeadline, checkIn, settings);
+            
+            Assert.True(originalDeadline.Policies.Select(p => p.FromDate)
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:39]: Please check "ToDate" too.
And maybe use a simple loop for more readability?

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                .SafeSequenceEqual(shiftedDeadline.Policies.Select(p => p.FromDate)));
+        }
+
+
+        [Fact]
+        public void Shift_should_change_by_exact_amount()
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:45]: by exact day count maybe? amount sounds not very appliable to days

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            var originalDeadline = new Deadline(FromDate);
+            var checkIn = FromDate;
+            var shiftSpan = TimeSpan.FromDays(-10);
+            var settings = new CancellationPolicyProcessSettings { PolicyStartDateShift = shiftSpan };
+
+            var shiftedDeadline = DeadlinePolicyProcessor.Process(originalDeadline, checkIn, settings);
+
+            Assert.Equal(originalDeadline.Date + shiftSpan, shiftedDeadline.Date);
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:54]: Think that it repeats the underlying logic too much.
At least an expression should contain a readable date.
The idea is to look at the test once and to allow check the logic by the eyes and calculator.
Test says that days must be shifted to 10 days?
Ok, let's get one date (show it), get the resulting date (show it) and check that it's really 10 days between them, without making calculations.
Please write to me if additional explanation needed

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+            Assert.Equal(FromDate, shiftedDeadline.Date);
+        }
+
+
+        private static readonly DateTime FromDate = new DateTime(2021, 1, 1);
+
+        private static readonly List<CancellationPolicy> Policies = new List<CancellationPolicy>
+        {
+            new CancellationPolicy(FromDate + TimeSpan.FromDays(1), 1d),
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:106]: Too much logic again

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs (hunk 1 above)
Comment (Raibo) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:39]: There is only `FromDate` in `CancellationPolicy`.
Simple loop - ok.

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs (hunk 2 above)
Comment (Raibo) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:45]: ok

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+using HappyTravel.EdoContracts.Accommodations.Internals;
+using HappyTravel.EdoContracts.Extensions;
+using Xunit;
+
+namespace HappyTravel.Edo.UnitTests.Tests.Services.Accommodations.Availability
+{
+    public class DeadlinePolicyProcessorTests
+    {
+        [Fact]
+        public void Positive_shift_should_be_made_zero()
```
[the remark this answers is not in the document]
Comment (Raibo) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs:15]: ok

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs (hunk 3 above)
Comment (Raibo) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs]: Added simple dates into test methods for easy reading.

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs (hunk 4 above)
Comment (Raibo) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/DeadlinePolicyProcessorTests.cs]: Changed to simple dates

**Review by mingazhev — APPROVED** [review]
Much easier and understand to me, thanks

**Pull request merged**